File: backend/app/models/file.py
import uuid

from sqlalchemy import Boolean, Column, DateTime, ForeignKey, Integer, String
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import relationship

from .base import Base


class File(Base):
    __tablename__ = "file"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)  # noqa: A003

    name = Column(String, nullable=False)
    path = Column(String, nullable=False)
    size = Column(Integer, nullable=False)
    encrypted = Column(Boolean, nullable=False)

    expiration = Column(DateTime, nullable=False)
    timestamp = Column(DateTime, nullable=False)

    user_id = Column(UUID(as_uuid=True), ForeignKey("user.id"), nullable=False)
    user = relationship("User", back_populates="files")

File: frontend/src/lib/types/File.ts
export class File {
    id: string;
    name: string;
    size: number;
    isEncrypted: string;
    timestamp: Date;
    expiration: Date;

    constructor(id: string, name: string, size: number, type: string, timestamp: Date, expiration: Date) {
      this.id = id;
      this.name = name;
      this.size = size;
      this.isEncrypted = type;
      this.timestamp = timestamp;
      this.expiration = expiration;
    }
}
